internal/go_repository.bzl

```diff
@@ -208,7 +208,24 @@ def _go_repository_impl(ctx):
         "GIT_SSL_CAINFO",
         "GIT_SSH",
         "GIT_SSH_COMMAND",
+        "GIT_CONFIG_COUNT"
     ]
+
+    # Git allows passing configuration through environmental variables, this will be picked
+    # by go get properly: https://www.git-scm.com/docs/git-config/#Documentation/git-config.txt-GITCONFIGCOUNT
+    if 'GIT_CONFIG_COUNT' in ctx.os.environ:
+        count = ctx.os.environ['GIT_CONFIG_COUNT']
+        if count:
+            if not count.isdigit or int(count) < 1:
+                fail("GIT_CONFIG_COUNT has to be a positive over 0 integer")
+            for i in range(int(count)):
+                key = 'GIT_CONFIG_KEY_%d' % i
+                value = 'GIT_CONFIG_VALUE_%d' % i
+                for j in [key, value]:
+                    if j not in ctx.os.environ:
+                        fail("%s is not defined as an environmetal variable, but you asked for GIT_COUNT_COUNT=%i" % (j, count))
```

Comment:
```suggestion
                        fail("%s is not defined as an environment variable, but you asked for GIT_COUNT_COUNT=%i" % (j, count))
```

Reply:
sounds good, I'll do that, maybe I cuold add a unit test for it to make sure it works as expected

Reply:
Unit tests for failures are unfortunately pretty tricky, so I won't make you do that. But manual test runs in the success and failure cases would be great.

Reply:
yeah I know, I've added a few tests to gazelle already, I'll do it as soon as I get sometime, don't expect this change to make it for this gazelle release anyway

Reply:
it's not unit tests for failures that are hard, it's unit test for repo rules are hard, didn't realize this is a repo rule, so no way I can write a unit test for it, I'll verify and keep you updated

Reply:
```
bazel-gazelle/tests/bcr$ GIT_CONFIG_COUNT=1 bazel build gazelle
INFO: Repository gazelle~override~go_deps~com_github_bazelbuild_buildtools instantiated at:
  callstack not available
Repository rule go_repository defined at:
  /home/mnaranjo/.cache/bazel/_bazel_mnaranjo/4fd2935ef7ec68b38be352a7828e3f9b/external/gazelle~override/internal/go_repository.bzl:313:32: in <toplevel>
INFO: Repository gazelle~override~go_deps~org_golang_x_mod instantiated at:
  callstack not available
Repository rule go_repository defined at:
  /home/mnaranjo/.cache/bazel/_bazel_mnaranjo/4fd2935ef7ec68b38be352a7828e3f9b/external/gazelle~override/internal/go_repository.bzl:313:32: in <toplevel>
ERROR: An error occurred during the fetch of repository 'gazelle~override~go_deps~com_github_bazelbuild_buildtools':
   Traceback (most recent call last):
        File "/home/mnaranjo/.cache/bazel/_bazel_mnaranjo/4fd2935ef7ec68b38be352a7828e3f9b/external/gazelle~override/internal/go_repository.bzl", line 226, column 29, in _go_repository_impl
                fail("%s is not defined as an environment variable, but you asked for GIT_COUNT_COUNT=%d" % (j, count))
Error in fail: GIT_CONFIG_KEY_0 is not defined as an environment variable, but you asked for GIT_COUNT_COUNT=1
ERROR: <builtin>: fetching go_repository rule @gazelle~override//:gazelle~override~go_deps~com_github_bazelbuild_buildtools: Traceback (most recent call last):
        File "/home/mnaranjo/.cache/bazel/_bazel_mnaranjo/4fd2935ef7ec68b38be352a7828e3f9b/external/gazelle~override/internal/go_repository.bzl", line 226, column 29, in _go_repository_impl
                fail("%s is not defined as an environment variable, but you asked for GIT_COUNT_COUNT=%d" % (j, count))
Error in fail: GIT_CONFIG_KEY_0 is not defined as an environment variable, but you asked for GIT_COUNT_COUNT=1
INFO: Repository gazelle~override~go_deps~org_golang_x_sync instantiated at:
  callstack not available
Repository rule go_repository defined at:
  /home/mnaranjo/.cache/bazel/_bazel_mnaranjo/4fd2935ef7ec68b38be352a7828e3f9b/external/gazelle~override/internal/go_repository.bzl:313:32: in <toplevel>
INFO: Repository gazelle~override~go_deps~com_github_pelletier_go_toml instantiated at:
  callstack not available
Repository rule go_repository defined at:
  /home/mnaranjo/.cache/bazel/_bazel_mnaranjo/4fd2935ef7ec68b38be352a7828e3f9b/external/gazelle~override/internal/go_repository.bzl:313:32: in <toplevel>
ERROR: /home/mnaranjo/.cache/bazel/_bazel_mnaranjo/4fd2935ef7ec68b38be352a7828e3f9b/external/gazelle~override/language/go/BUILD.bazel:15:11: @gazelle~override//language/go:go depends on @gazelle~override~go_deps~com_github_bazelbuild_buildtools//build:go_default_library in repository @gazelle~override~go_deps~com_github_bazelbuild_buildtools which failed to fetch. no such package '@gazelle~override~go_deps~com_github_bazelbuild_buildtools//build': GIT_CONFIG_KEY_0 is not defined as an environment variable, but you asked for GIT_COUNT_COUNT=1
ERROR: Analysis of target '//:gazelle' failed; build aborted: 
INFO: Elapsed time: 0.616s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded, 0 targets configured)
    Fetching repository @gazelle~override~go_deps~com_github_pmezard_go_difflib; starting
```

Reply:
I fixed a bug in while testing, turns out count was a string and not an int and %d was making it fail, now I believe we're fine to get a final review and merge unless there are further reviews. @fmeum